### PR TITLE
Added template-spec example to module usage

### PR DESCRIPTION
**Pull request by MrMCake**
# Change

- Added template-spec example to module usage

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update (Wiki)

### CODE REVIEW

File: docs/wiki/ModulesUsage.md [hunk 1]
```diff
@@ -6,18 +6,26 @@ This section gives you an overview of how to use the bicep modules.
...
 ---
 
-## Deploy local template
+## Deploy template
+
+This section shows you how to deploy a bicep template.
+
+### Deploy local template
+
+This sub-section givens you an example on how to deploy a template on your local drive.
```
Comment (MariusStorhaug) [inline docs/wiki/ModulesUsage.md:26]: typo
givens -> gives
on -> from

```md
- This sub-section givens you an example on how to deploy a template on your local drive.
+ This sub-section **gives** you an example on how to deploy a template **from** your local drive.
```

File: docs/wiki/ModulesUsage.md [hunk 2]
```diff
@@ -62,7 +72,7 @@ $inputObject = @{
 New-AzResourceGroupDeployment @inputObject
```
Comment (MariusStorhaug) [inline docs/wiki/ModulesUsage.md:72]: I know this is an example, but should TemplateUri = ... Azure/ResourceModules ?

File: docs/wiki/ModulesUsage.md (hunk 1 above)
Comment (MrMCake) [reply docs/wiki/ModulesUsage.md]: fixed

File: docs/wiki/ModulesUsage.md (hunk 2 above)
Comment (MrMCake) [reply docs/wiki/ModulesUsage.md:72]: It should. Update both local & remote

File: docs/wiki/ModulesUsage.md [hunk 3]
```diff
@@ -75,3 +85,115 @@ $inputObject = @(
...
+
+This section shows you how you can orchestrate a deployment using multiple resource modules
+
+### Template-orchestration
+
+The _template-orchestrated_ approach means using a _main_ or so-called _master template_ for deploying resources in Azure. The _master template_ will only contain nested deployments, where the modules – instead of embedding their content into the _master template_ – will be linked from the _master template_.
+
+With this approach, modules need to be stored in an available location, where Azure Resource Manager (ARM) can access them. This can be achieved by storing the modules templates in an accessible location location like _template specs_ or the _bicep registry_.
+
+In an enterprise environment, the recommended approach is to store these templates in a private environment, only accessible by enterprise resources. Thus, only trusted authorities can have access to these files.
```
Comment (MariusStorhaug) [inline docs/wiki/ModulesUsage.md:99]: Do we imply here that customers should fork our repo and deploy it to their own env?

Comment (MrMCake) [reply docs/wiki/ModulesUsage.md:99]: It's one of the 3 scenarios with decribe in the `Getting started` section. 
It's supposed to imply that an enterprise should, if they want to use the repo (including the pipelines) not publish their templates to a public space like the public bicep registry (once available)

**Review by MariusStorhaug — APPROVED** [review]
All comments addressed. Approved

**Pull request merged**